pkg/util/pool/bucketed_pool.go

```diff
@@ -42,6 +42,9 @@ func NewBucketedPool[T ~[]E, E any](maxSize uint, makeFunc func(int) T) *Buckete
 }
 
 // Get returns a new slice with capacity greater than or equal to size.
+// If no bucket large enough exists, a slice larger than the requested size
+// of the next power of two is returned.
+// Get guarantees the resulting slice always has a capacity in power of twos.
```

Comment:
I think this could be a little clearer:

```suggestion
// The resulting slice always has a capacity that is a power of two.
// If size is greater than maxSize, then a slice is still returned, however it may not be drawn from a pool.
```